### DISCUSSION TITLE
Quick review of 2.0a5

### DISCUSSION THREAD

**Discussion by Every0ne** [automadcms/discussions#13] [General]
I'm very impressed by the new AM 2. I've migrated a working website to 2.0a5 to see how it would work out. These are my thoughts so far.

1. Plugins from 1.10 that I wrote for this specific website were throwing unrecoverable errors in PHP. I expected breaking changes in the API, but there doesn't seem to be documentation available for 2 yet, or did I miss it?
2. I really like how the new version of Editor.js is so much snappier and responsive. What I don't like, and what makes it a showstopper for me, is how some (all?) text styling is now applied as inline style attribute. In the old days it was possible, in example, to apply centered text to a paragraph, then overwrite it in mobile view, when needed, because it was applied with a class. With how the new editor works now, such overrides cannot be applied, at leas not easily nor reliably. It's still possible to apply classes for paragraph blocks by typing them in, but this basically means that it's needed to define and document text formatting classes for stuff that the editor seemingly provides. Explaining to somebody why can't they use editor's functionality to justify text, but type in classes instead would be a nightmare. Is it possible to get back to how styling of paragraphs worked?
3. Clicking _Open readme file of current theme_ reopens the dashboard in a new tab instead. Does not happen with standard theme tho, dunno why.
4. WebP is finally recognized, but support for AVIF still missing... Will it it arrive in the final 2.0?
5. Seems that page format went fully to JSON. I liked how easy it was to edit pages just in notepad, but since Editor.js saves its data this way, plus I guess there aren't many people who edit pages in notepad, then this move makes total sense. I know that JSON is text and can be still edited in notepad, but It's harder on the eyes and requires significantly more effort and attention to not break the page. I agree that this change makes parsing page files easier. I'll just miss the old ways a bit.

These are the nits that I picked, apart from which everything else looks like a huge leap forward. Marc, thank you for your monumental amount of continued inspiring work on AM. It has become a foundation for the websites that I make for both myself and friends. Remember to not get discouraged by selfserving aƨƨholes that demand stuff.

**Comment by marcantondahmen** [comment]
Hey, thank you so much for this detailed feedback! This is extremely useful and much appreciated. I will carefully look at the points whenever there is some time. Especially point 2.

Regarding JSON, using a standard simplifies a lot and the parser is more robust and mature. The focus has shifted a bit towards UI and away from editing files directly. Also features such as drafts benefit a lot from a more flexible format.
